### DISCUSSION TITLE
Referencing SLO in Dashboard when deploying new project

### DISCUSSION THREAD

**Discussion by diablovos** [Dynatrace/dynatrace-configuration-as-code#373] [Q&A]
I am trying to deploy a new application on Dynatrace with monaco (management-zone, application-web, app-detection-rule, slo, dashboard ....) and i am facing an issue with referencing the created SLOs in the Dashboard.

In my dashboard.yaml i set the slo id :

config:
  - dashboard: "overview-dashboard.json"

dashboard:
  - name: "/templates/default/management-zone/management-zone.name"
  - MzId: "/templates/default/management-zone/management-zone.id"
  - sloX: "/templates/default/slo/service_slo.id"
  - appId: "/templates/default/application-web/application.id"

and i already have  an slo directory with slo.yaml  :

config:
  - service_slo: "slo_service.json"

service_slo:
  - app: "/templates/default/management-zone/management-zone.name"
  - name: "ServiceSLO"
  - entity: "serviceName_1"
  - type: "SERVICE"

 
the problem is that the slo id is returned as an url : https://xxxxxxx/api/v2/slo/ccf0a886-4543-3d71-96b8-30ad6eaa84ea
and so slo tile is not working well in my dashboard.

Any idea how can i resolve this?

**Comment by tatomir146** [comment]
@diablovos I have just tested this and haven't found any problem. You are referencing SLO with `assignedEntities` parameter in dashboard JSON? Which version of cluster are you using?

**Reply by diablovos** [reply]
Hi @tatomir146   we are using Dynatrace Managed, version 1.218 (soon we will be moving to 1.220)

here is my dashboard.json :

{
  "dashboardMetadata": {
    "name": "{{.name}} Overview",
    "shared": true,
    "owner": "xxxxxxxxxxxxxxx",
    "sharingDetails": {
      "linkShared": true,
      "published": true
    },
    "dashboardFilter": {
      "timeframe": "",
      "managementZone": null
    }
  },"tiles": [
  {
   **"assignedEntities": [
    "{{ .slo0 }}"**
   ],
   "bounds": {
    "height": 152,
    "left": 38,
    "top": 608,
    "width": 304
   },
   "configured": true,
   "name": "Service-level objective",
   "tileFilter": {
    "managementZone": null,
    "timeframe": "-1w"
   },
   "tileType": "SLO"
  }
}
 ]
}


and my dashboard.yaml : 


config:
  - dashboard: "overview-dashboard.json"

dashboard:
  - name: "/templates/ODM/management-zone/management-zone.name"
  - MzId: "/templates/ODM/management-zone/management-zone.id"
  **- slo0: "/templates/ODM/slo/service_slo0.id"**

**Reply by diablovos** [reply]
Any answer please!!!

**Reply by tatomir146** [reply]
Hi @diablovos, I see you are combining values for "name" parameter in JSON and this can't work. Name can *only* contain a `{{ .name }}` variable, otherwise monaco won't be able to find and update dashboard. Please try fixing this and then updating dashboard again. More details here: https://dynatrace-oss.github.io/dynatrace-monitoring-as-code/configuration/configuration_structure#configuration-yaml-structure
